test/steps/steps.ts

```diff
@@ -12,48 +13,53 @@ interface ICredentials {
-    // //    await auth.context.storageState(storedState);
-    // //    //await homePage.openHomePage();
-    //     //const location = `../../playwright/.auth/${authFile}.json` // /user.json';
-    //     //const storedState = require(location);
-    //     //auth.context = await browser.newContext( { storageState: storedState } );
-    //     const storedState = JSON.parse(fs.readFileSync(filePath, 'utf-8'));
-    //     auth.context = await browser.newContext( { storageState: storedState } );
-// 
-    //     //await homePage.page.context().storageState(storedState);
-    //     //await homePage.openHomePage();
-  // 
+    auth.context = await browser.newContext( { storageState: filePath } );
+    auth.page = await auth.context.newPage();
+    pages.homePage = new Pages.HomePage(auth.page);
+
+    //    console.log('Storage state file exists. Loading state from it.');
+    //    const storedState = JSON.parse(fs.readFileSync(`${filePath}`, 'utf-8'));
+    //    auth.context = await page.context(); // Initialize auth.context
+    //    await auth.context.storageState(storedState);
+    //    //await homePage.openHomePage();
+        //const location = `../../playwright/.auth/${authFile}.json` // /user.json';
+        //const storedState = require(location);
+        //auth.context = await browser.newContext( { storageState: storedState } );
+        // const storedState = JSON.parse(fs.readFileSync(filePath, 'utf-8'));
+        // auth.context = await browser.newContext( { storageState: filePath } );
+
+        //await homePage.page.context().storageState(storedState);
+        //await homePage.openHomePage();
+  
     // } else {
-        console.log('Storage state file does not exist. Proceeding with login.');
-        await loginPage.openLoginPage();
-        await loginPage.enterUser(credential.username);
-        await loginPage.enterPassword(credential.password);
-        await loginPage.clickOnSignIn();
-        await homePage.assertOpened()//assert login finished
-        // Save the storage state to a file
-        const storageState = await homePage.page.context().storageState()  //auth.context.storageState();
-       // auth.context = await homePage.page.context();
-        fs.writeFileSync(filePath, JSON.stringify(storageState));
-        console.log(`Storage state saved to ${filePath}.`);
+
    // }
 });
 
-Then('I see I am on the home page',async ({homePage}) => {
-    await homePage.assertOpened();
+Then('I see I am on the home page',async ({ pages }) => {
+    await pages.homePage.assertAuthorized();
 })
 
-When('I go to home page', async ({homePage, auth}) => {
-    const authFile = path.resolve(__dirname, `../../playwright/.auth/${auth.userName}.json`);
-    const storedState = JSON.parse(fs.readFileSync(authFile, 'utf-8'));  // require(authFile);
-    homePage.page = await auth.context.newPage();// How do I pass the retrieved storage state to the active page?
```

Comment:
> How do I pass the retrieved storage state to the active page?

When page is already initialized, Playwright does not allow to switch to new state.
That's why we use need to use `pages` fixture that initializes `homePage` lazily after login step.